core/src/banking_stage.rs

```diff
@@ -1202,6 +1213,129 @@ impl BankingStage {
         }
     }
 
+    #[allow(clippy::too_many_arguments)]
+    fn commit_transactions(
```

Comment:
Love to see someone break these monster functions up. Might take it one step further, this function is still fairly large and seems to be doing 3 three distinct things:
1. commiting transactions
2. collecting commit statuses from commit results (1)
3. collect balances and send from commit results (1)

Can either rework this PR or do a follow up PR if you agree - could potentially do something like:
```
fn commit_transactions(...) -> Vec<CommitTransactionDetails>;
fn get_commit_status(tx_results) -> Vec<CommitTransactionDetails>;
fn collect_balances_and_send_status_batch(...);

fn commit_and_send_status_batch() -> (commit_time_us, Vec<CommitTransactionDetails>); // Combination of the above 3 functions to replace this one
```

Reply:
yea, it's a good idea. I think a follow-up would be the best way.